internal/resource/protoc-gen-resource-types/main.go

```diff
@@ -12,7 +12,7 @@ import (
 )
 
 var (
-	file = flag.String("file", "-", "where to load data from")
+	prefix = flag.String("prefix", "resource_types.gen", "filename prefix to use. '.go' will be appended to this to create the full filename")
```

Comment:
Why?

* proto/private/pbdemo/{v1,v2} packages required annotations to have code gen run so they could be registered as resource types
* `mog` also runs on the proto/private directory and generates files like `*.gen.go`
* After running `buf` but before running `mog` we remove all `*.gen.go`
* This meant that code gen would run but then our protobuf post processing would wipe it out.
* Therefore I needed to allow these files to be output using an alternative name just in this directory but not proto-public.